### ISSUE TITLE
Copy markdown image button not working when uploading a new photo for a post

### ISSUE THREAD

**Issue by simonxabris** [thepracticaldev/dev.to#7251]
**Describe the bug**

Hi!

I tried to create my first post on DEV and when I tried to upload a picture and copy the generated markdown URL, the text was just simply not copied to my clipboard. I tried it in basically every major browser (Firefox, Safari, Chrome, Edge) and it did not work in any of them. However I tried this out on my phone too (iPhone, safari) and it worked as expected.

I found two issues that I think are related to this.

[This one](https://github.com/thepracticaldev/dev.to/issues/6169) is definitely related, because it's the same UI where I experienced the issue.

and [this one](https://github.com/thepracticaldev/dev.to/issues/7109) might be related since it's about copying as well, but at a different part of the app.

I've opened this new issue, because the first one I referenced makes it seems like it's platform specific (it has area:mobile label on it), while in my experience it is not tied to platform. So this one could be a more generic and possible solve the problem mentioned in the issues I referenced.

**To Reproduce**

- Start writing a new article.
- Upload a picture.
- Click on the copy icon next to the generated markdown url text.
- Check if it actually copied the text.

**Expected behavior**

The expected behavior is to actually copy the markdown text to the user's clipboard when the button is clicked.

**Desktop (please complete the following information):**

- OS: macOS 10.14.6
- As I stated above, I tested this in multiple browsers and it did not work in any of them

**Smartphone (please complete the following information):**

- Device: iphone 8
- OS: ios 13.4.1
- Browser: safari
- Version: 13

**Additional context**

So I went ahead and searched fo the bug in the code and I think I found it. The bug is in this function: https://github.com/thepracticaldev/dev.to/blob/80737b540453afe8775128cb37bd379b7c09c7e8/app/javascript/article-form/elements/imageManagement.jsx#L77

The command to actually copy the text is only called on iOS devices for some reason.

I'd be happy to submit a PR with a fix if this is something you would be interested in!

**Comment by rhymes** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @thepracticaldev/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by rhymes** [comment]
Hi @simonxabris, thanks for investigating! I can definitely reproduce the error on Firefox 76 on desktop. Also, please do submit a PR if you'd like. I'm going to assign you the issue :)

**Comment by simonxabris** [comment]
Thanks @rhymes . I think i'll have time to work on this at the end of the week. If I have any questions in the process I'll ask it here.

**Comment by rhymes** [comment]
Perfect, thank you @simonxabris!

**Comment by fdoxyz** [comment]
Hello @simonxabris thanks for taking a swing at this! Did you had a chance to try sort out the issue?

I was implementing a fix for the Android side (mostly Native App) but had to rearrange some stuff on this exact method you mentioned. I changed the API used in order to “copy to the clipboard” for all other browsers. Would you be willing to take a look at [this PR](https://github.com/thepracticaldev/dev.to/pull/7382) and see if it fixes the problem you were experiencing? 

I would appreciate your input in any way since I know you were working on this. If you had something going here I'd be happy to rebase after you. Let me know if that PR does the trick though or if you need help with anything here or on that PR too :)

**Comment by simonxabris** [comment]
Hi @fdoxyz 

Yeah I tried but I actually couldn't get the project to run locally so I've practically made zero progress regarding this issue 😅. 

I've looked at your PR and it's pretty much what I would've done (without the android stuff). My only concern is that the `clipboard.writeText` API is not supported in in desktop and ios safari (and also explorer) [according to caniuse](https://caniuse.com/#feat=mdn-api_clipboard_writetext). I've seen that you seem to handle this in the `catch` of the promise, but I'm not 100% sure the promise is going to throw an error if the API is actually unsupported.

I would happily test your changes but as I said I could not get the project up and running and I won't have time today to try again. If you could test this in safari to see if it works (if you already did I'm sorry 😃) I think we can merge your PR and close this issue with it. If you don't have a device which you could test it on I will ask for some help regarding the local startup of the project and try to test it myself.

**Comment by fdoxyz** [comment]
No worries @simonxabris! Feel free to ping me if ever stuck somewhere in the installation/setup process, I'd be happy to help. Of course no pressure with having to test the PR though.

Thanks for the heads up with the Clipboard API not working if unsupported by the browser! I had tried a few browsers & devices but it did fail in Safari and I hadn't tried that. I just took care of it on another commit. That was a great call, thanks 😃
